services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceControllerClusterConfig.java

```diff
@@ -894,7 +904,13 @@ public VeniceControllerClusterConfig(VeniceProperties props) {
     this.emergencySourceRegion = props.getString(EMERGENCY_SOURCE_REGION, "");
     this.allowClusterWipe = props.getBoolean(ALLOW_CLUSTER_WIPE, false);
     this.concurrentInitRoutinesEnabled = props.getBoolean(CONCURRENT_INIT_ROUTINES_ENABLED, false);
-    this.controllerInAzureFabric = props.getBoolean(CONTROLLER_IN_AZURE_FABRIC, false);
+    this.controllerClusterHelixCloudEnabled = props.getBoolean(CONTROLLER_CLUSTER_HELIX_CLOUD_ENABLED, false);
+    this.controllerStorageClusterHelixCloudEnabled =
+        props.getBoolean(CONTROLLER_STORAGE_CLUSTER_HELIX_CLOUD_ENABLED, false);
+    this.controllerHelixCloudProvider = props.getString(CONTROLLER_HELIX_CLOUD_PROVIDER, "");
```

Comment:
Let's make this into a `CloudProvider` object one time instead of doing `.toUpperCase` and `.valueOf` every time we read this value?

Reply:
That would require moving the exception to `VeniceControllerClusterConfig`. Is that ok?

If not, I can definitely move the toUpperCase there.

Reply:
Yes, that's okay. It is better that way actually so that we fail on host startup instead of waiting to fail till we need to use it